### PR TITLE
Add slack integration for Kilonova candidates

### PR DESCRIPTION
**Pull request by JulietteVl**

### CODE REVIEW

File: fink_filters/filter_kn_candidates/filter.py [hunk 1]
```diff
@@ -2,13 +2,17 @@
 from pyspark.sql.types import BooleanType
 
 import pandas as pd
+import requests
+import os
 
 @pandas_udf(BooleanType(), PandasUDFType.SCALAR)
-def kn_candidates(knscore, drb, classtar, jd, jdstarthist, ndethist, cdsxmatch) -> pd.Series:
+def kn_candidates(objectId, knscore, drb, classtar, jd, jdstarthist, ndethist, cdsxmatch) -> pd.Series:
     """ Return alerts considered as KN candidates
```
Comment (JulienPeloton) [inline fink_filters/filter_kn_candidates/filter.py:10]: Could you update the docstring to specify that alerts are also sent via Slack if the environment variable `KNWEBHOOK` is defined?

File: fink_filters/filter_kn_candidates/filter.py [hunk 2]
```diff
@@ -62,5 +66,13 @@ def kn_candidates(knscore, drb, classtar, jd, jdstarthist, ndethist, cdsxmatch)
 
     f_kn = high_knscore & high_drb & high_classtar & new_detection
     f_kn = f_kn & small_detection_history & cdsxmatch.isin(keep_cds)
+    
+    for alertID in objectId[f_kn]:
+        slacktext = f'new kilonova candidate alert: \n<http://134.158.75.151:24000/{alertID}>'
+        requests.post(
+            os.environ['KNWEBHOOK'],
```
Comment (JulienPeloton) [inline fink_filters/filter_kn_candidates/filter.py:73]: Here we need to be careful with error handling -- if `KNWEBHOOK` is not defined, then the code will crash with `KeyError`:

```python
In [1]: import os

In [2]: os.environ['toto']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-329d8044c191> in <module>()
----> 1 os.environ['toto']

/opt/anaconda3-5.2.0/lib/python3.6/os.py in __getitem__(self, key)
    667         except KeyError:
    668             # raise KeyError with the original key value
--> 669             raise KeyError(key) from None
    670         return self.decodevalue(value)
    671

KeyError: 'toto'
```

So we need to rephrase differently. I propose (not tested - just for getting the idea):

```python
import logging

# ...

# in the  filter function
if 'KNWEBHOOK' in os.environ:
  # then you have the loop
else:
  log = logging.Logger('Kilonova filter')
  log.warning('KNWEBHOOK is not defined as env variable -- the message has not been sent to Slack')
```

**Review by JulienPeloton — COMMENTED** [review]
Thanks @JulietteVl ! I left a comment on error handling (if the env variable is not defined)

File: fink_filters/filter_kn_candidates/filter.py (hunk 2 above)
Comment (JulietteVl) [reply fink_filters/filter_kn_candidates/filter.py:73]: I gave it a try in jupyter, I see the log warning when I run the commands on the driver, but not when it runs on the executors.
Is there a way for the users to access the warning message?

**Review by JulienPeloton — APPROVED** [review]

**Pull request merged**